servicetalk-encoding-netty/src/main/java/io/servicetalk/encoding/netty/NettyChannelContentCodec.java

```diff
@@ -177,6 +182,11 @@ public void onComplete() {
                 });
     }
 
+    @Override
+    public Buffer decode(final Buffer src, final BufferAllocator allocator) {
+        return decode(src, 0, src.readableBytes(), allocator);
```

Comment:
Keep the `AbstractZipContentCodec` as it's right now in this PR, but for `NettyChannelContentCodec` consider doing the opposite: implement `decoder(Buffer, BufferAllocator)` and do the slice in this overload.